src/software/simulation/simulator_robot.cpp

```diff
@@ -359,7 +360,21 @@ void SimulatorRobot::onDribblerBallContact(PhysicsRobot *physics_robot,
 {
     if (dribbler_rpm > 0)
     {
-        applyDribblerForce(physics_robot, physics_ball);
+        auto iter =
+            std::find_if(balls_in_dribbler_area.begin(), balls_in_dribbler_area.end(),
+                         [physics_ball](DribblerBall dribbler_ball) {
+                             return dribbler_ball.ball == physics_ball;
+                         });
+
+        if (iter == balls_in_dribbler_area.end())
+        {
+            throw std::runtime_error("Trying to dribble ball not in the dribbler area");
```

Comment:
Fancy codecov warning: 
```
Check warning on line 371 in src/software/simulation/simulator_robot.cpp

Codecov / codecov/patch

src/software/simulation/simulator_robot.cpp#L371

Added line #L371 was not covered by tests

```

Reply:
Fancy. Unfortunately I can't test this line with the current public interface because getting here should be impossible.

This function only gets called during each physics step while there is a contact between a ball and dribbler. This also happens _after_ the `startContact` callback. Therefore since the `startContact` callback is always called first, a ball should exist in the `balls_in_dribbler_area` vector. And since this function is only called when there is a contact between the ball and dribbler, the physics_ball should always be valid and exist in the vector.

There only way I can think of to cover that line would be to call the `onDribblerBallContact` function directly with an invalid `physics_ball`, but the function is private.